apps/meteor/app/lib/server/methods/getMessages.ts

```diff
@@ -21,6 +21,20 @@ Meteor.methods({
 			throw new Meteor.Error('error-not-allowed', 'Not allowed', { method: 'getSingleMessage' });
 		}
 
+		const prids = msgs.reduce<string[]>((prids, m) => {
+			const room: IRoom = Rooms.findById(m.rid);
```

Comment:
please move this find to outside the loop/reduce. we should avoid multiple round trips to db as much as possible.

you already have the list of all room ids you need to query, the `rids` variable from line 19. you can use to perform a single find, returning only the `prid` field, something like:

```
const rooms = Rooms.findByIds(rids, { fields: { prid: 1 } }).fetch();
```

Reply:
updated! Sorry for the miss on that from my side